index.js

```diff
@@ -1,26 +1,152 @@
 // 1.Fibonacci
 
-// function fibo(n) {
-//   return n <= 1 ? n : fibo(n - 1) + fibo(n - 2);
-// }
-// fibo(5);
-// fibo(6);
-// fibo(7);
+function fibo(n) {
+  return n <= 1 ? n : fibo(n - 1) + fibo(n - 2);
+}
+fibo(5);
+fibo(6);
+fibo(7);
 
 // 2. Implementing with reduce
 
 //concat;
+const arr = [1, 2];
+const arr1 = [3, 4];
+const arr2 = 5;
+
+const concatFn = (...array) =>
+  array.reduce((array, newEl) =>
+    array.constructor === Array ? [...array, ...newEl] : [...array, newEl]
+  );
+
+concatFn(arr, arr1);
+concatFn(arr1, arr2);
+console.log(`My new ${arr}`);
+
+//filter and map;
+
+let friends = [
+  {
+    name: "John",
+    age: 15,
+  },
+  {
+    name: "Michael",
+    age: 25,
+  },
+  {
+    name: "Lisa",
+    age: 30,
+  },
+];
+
+const filter = (newArr, fn) => {
+  return newArr.reduce(function (acc, person) {
+    if (fn(person)) {
+      return [...acc, person];
+    }
+    return acc;
+  }, []);
+};
+
+let fn1 = (obj) => {
+  return obj.age > 18;
+};
+
+console.log(filter(friends, fn1));
 
-//filter;
 // find;
+let friends = [
+  {
+    name: "John",
+    age: 15,
+  },
+  {
+    name: "Michael",
+    age: 25,
+  },
+  {
+    name: "Lisa",
+    age: 30,
+  },
+];
+const findLisa = (newArr, fn) => {
```

Comment:
you don't need to return array here, you return element or if element is abest - null || undefined